rclcpp/src/rclcpp/qos.cpp

```diff
@@ -67,7 +69,15 @@ KeepAll::KeepAll()
 
 KeepLast::KeepLast(size_t depth)
 : QoSInitialization(RMW_QOS_POLICY_HISTORY_KEEP_LAST, depth)
-{}
+{
+  if (depth == 0) {
+    RCLCPP_WARN_ONCE(
+      rclcpp::get_logger(
+        "rclcpp"),
+      "A zero depth with KEEP_LAST doesn't make sense; no data could be stored."
+      "This will be interpreted as SYSTEM_DEFAULT");
+  }
+}
```

Comment:
I would rather warn in `QoS::QoS()` and `QoS::keep_last()`, so we cover all the cases.

Reply:
Good point.  I've actually *added* it to both `QoS::QoS` and to `QoS::keep_last`, because it is possible to construct a `KeepLast` object without going through those APIs.